[FIX] repair: correct field and apply OR logic in search

elct-odoo opened the pull request:
The _search_date_category method was using the incorrect field 'scheduled_date' instead of 'schedule_date', which broke the domain logic. This commit also wraps individual domain clauses in an expression.OR(...) to ensure proper filtering across multiple date categories, in line with the updated ORM search behavior (task-4150178).

Caused by:
https://github.com/odoo/odoo/commit/92301a5b300dec1ddfca44dc35318b83d67c56fa

This is the same fix applied to _search_date_category in mrp & stock.

https://github.com/user-attachments/assets/e45791c3-feaf-4d52-9f63-6ff43fb84ca8

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/222256.png)](https://mergebot.odoo.com/odoo/odoo/pull/222256)

elct-odoo commented:
> Hello @elct-odoo your fix was fine, but I adjusted the commit message to match the [expected bugfix commit message steps to reproduce](https://github.com/odoo/odoo/wiki/Contributing#reporting-issues) and the [guideline of explain why you changed something, not what you changed in the code (since we can already see that)](https://www.odoo.com/documentation/18.0/contributing/development/git_guidelines.html#commit-message-full-description), including putting the opw at the bottom of the commit message. Please ensure you follow those guidelines for your next bugfix PRs as it takes us a lot of time/mental energy to go back and forth with PRs over small changes like that.
> 
> 
> 
> I also modified your test to make it a bit more realistic + more robust to avoid failing if the test is run around midnight when "`now()`" may have the same day as "`'today'`".
> 
> 
> 
> @robodoo r+

Will do, I apologize for the inconvenience.

Thank you!
